portal-bridge/src/types/state/execution.rs

```diff
@@ -164,10 +174,13 @@ impl State {
                     self.database.accounts.insert(beneficiary, account);
                 }
             }
-            updated_accounts.insert(beneficiary);
         }
 
-        if self.get_root()? != block_tuple.header.header.state_root {
+        let RootWithTrieDiff {
+            root,
+            trie_diff: changed_nodes,
+        } = self.get_root_with_changed_nodes()?;
```

Comment:
There is one thing that crossed my mind that worries me because I'm not sure how `eth_trie` works.

Are we sure that `get_root_with_changed_nodes` doesn't return nodes that didn't change? (e.g. they were just read during transaction but not modified)

Reply:
I don't read from the trie, I only update/delete accounts. For reading during evm execution I interface with the accounts in a flat key value store. 

> they were just read during transaction

This interaction isn't done at least in how I implemented the code, so in this case it would only return updated nodes. Maybe someone could do X, but I didn't read or implement using X. How I implemented things we should be good.